### PR TITLE
#150: regex-matcher added

### LINKED ISSUE #150: to implement `matches-regex` matcher
According to `StringRegularExpression.matchesRegex()`: https://hamcrest.org/JavaHamcrest/javadoc/2.1/org/hamcrest/core/StringRegularExpression.html#matchesRegex-java.lang.String-

### PR DESCRIPTION
**Pull request by includealex**
Closes #150

### CODE REVIEW

**Comment by includealex** [comment]
@Graur please take a look.

File: src/test/eo/org/eolang/hamcrest/failed-output/gen-by-regex-failed-output.eo [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+[] > gen-by-regex-failed-output
+  [] > suggestion
+    assert-that > @
+      "abc"
+      $.gen-by-regex
+        "/($[0-9])/"
+  assert-that > @
+    suggestion
+    $.equal-to "\nExpected: generated by the regular expression </($[0-9])/>\n     but: </($[0-9])/> could not generate given string"
```
Comment (Graur) [inline src/test/eo/org/eolang/hamcrest/failed-output/gen-by-regex-failed-output.eo:37]: @includealex It should be something like: "Expected: a string matching the pattern </($[0-9])/>
     but: the string was <abc>"

File: src/main/eo/org/eolang/hamcrest/matchers/gen-by-regex-matcher.eo [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
++alias org.eolang.txt.regex
++alias org.eolang.txt.sprintf
++home https://github.com/objectionary/eo-hamcrest
++package org.eolang.hamcrest.matchers
++rt jvm org.eolang:eo-hamcrest:0.0.0
++version 0.0.0
+
+# Checks if given string can be generated
+# by the regular expression.
+[str] > gen-by-regex-matcher
```
Comment (Graur) [inline src/main/eo/org/eolang/hamcrest/matchers/gen-by-regex-matcher.eo:32]: @includealex We are following the original Hamcrest library name convention. So can you please rename this object to `matches-regex-matcher`?

File: src/main/eo/org/eolang/hamcrest/assert-that.eo [hunk 3]
```diff
@@ -148,3 +149,8 @@
   [str] > string-starts-with
     string-starts-with-matcher > @
       str
+
+  # Tests if the argument can be generated by the given regular expression.
+  [str] > gen-by-regex
```
Comment (Graur) [inline src/main/eo/org/eolang/hamcrest/assert-that.eo:154]: @includealex We are following the original Hamcrest library names convention. So can you please rename this object to `matches-regex`?

**Review by Graur — CHANGES_REQUESTED** [review]
@includealex Good one, just a few comments below:

**Comment by includealex** [comment]
@Graur please take a look.

**Review by Graur — CHANGES_REQUESTED** [review]

**Review by Graur — APPROVED** [review]
@includealex Thanks!

**Comment by Graur** [comment]
@rultor merge

**Pull request merged**